File: others/comp9041-Q/cs9041/lab07/count_word.pl
#!/usr/bin/perl -w
$ori_word = shift @ARGV;
# print $ori_word;
$word = uc $ori_word;
# print $word,"\n";
$count = 0;
while($line = <>){
	@arr = $line =~ /[a-zA-Z]+/g;
	# $tmp = 0;
	foreach $x (@arr){
		chomp $x;

		$up = uc $x;
		# print $x,",",$up,"|";
		if ($up eq $word){
			$count++;
		}
	}
	# $count += $tmp;
	# print $tmp," $count";
	# print join":",@arr,"\n";
}
print "$ori_word occurred $count times","\n";

File: others/comp9041-Q/cs9041/lab07/count_word.py
#!/usr/bin/python3
import re,sys
sum = 0
word = sys.argv[1]
for line in sys.stdin:
    l = re.split(r'[^a-zA-Z]',line)
    new_l = []
    for x in l:
    	if x.lower() == word.lower():
    		new_l.append(x)
    sum += len(new_l)
print("{} occurred ".format(word),end='')
print(sum,end='')
print(" times")